docker/postgres_upgrade.sh

```diff
@@ -0,0 +1,165 @@
+    echo -n "."
+    sleep 1
+done
+
+POSTGRES_FULL_VERSION=$(docker-compose exec -T db-postgres psql --user=seed -d postgres -c 'SELECT version();' | sed -n 3p)
+# Extract the version number using regex
+POSTGRES_VERSION=$(echo "$POSTGRES_FULL_VERSION" | sed -n 's/.*PostgreSQL \([0-9]*\.[0-9]*\).*/\1/p')
+# Capture the TimescaleDB version from the container using SQL command
+TIMESCALE_VERSION=$(docker-compose exec -T db-postgres psql --user=seed -d postgres -t -A -c "SELECT extversion FROM pg_extension WHERE extname='timescaledb';")
+
+
+echo "Disconnecting all other users from the 'seed' database..."
+docker-compose exec -T db-postgres psql --user=seed -d postgres -c "SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname = 'seed';"
+export PG12_DUMP="/share/seed-pg${POSTGRES_VERSION}-${TIMESCALE_VERSION}-$(date +%Y%m%d%H%M%S).dump"
+echo "Creating PostgresV: ${POSTGRES_VERSION} TimescaleV: ${TIMESCALE_VERSION} dump file of $PG12_DUMP..."
+docker-compose exec -T db-postgres pg_dump --username=seed --dbname=seed --verbose --no-owner --no-acl -Z7 -Fc -f "${PG12_DUMP}"
+
+
+echo "============================================"
+echo "Starting PostgreSQL 13 Initialization Process"
+echo "============================================"
+echo "Waiting for PostgreSQL 13 to start..."
+# Wait until PostgreSQL starts and is ready to accept connections.
+until docker-compose exec -T seed-pg13 pg_isready -U seed -d seed > /dev/null 2>&1; do
+    echo -n "."
+    sleep 1
+done
+
+echo
+echo "PostgreSQL 13 is up and running."
+echo
+
+echo "Checking initial PostgreSQL and TimescaleDB versions for PostgreSQL 13..."
+# Show PostgreSQL version
+docker-compose exec -T seed-pg13 psql --user=seed -d postgres -c 'SELECT version();'
+
+# Show TimescaleDB version
+docker-compose exec -T seed-pg13 psql --user=seed -d postgres -c '\dx timescaledb'
+
+echo "Setting up database and extensions on PostgreSQL 13..."
+# Now, attempt to drop the 'seed' database
+docker-compose exec -T seed-pg13 psql --user=seed -d postgres -c 'DROP DATABASE IF EXISTS "seed";'
+
+# Create the "seed" database with the owner "seed"
+docker-compose exec -T seed-pg13 psql --user=seed -d postgres -c 'CREATE DATABASE "seed" WITH OWNER = "seed";'
+
+# Grant all privileges on the "seed" database to the user "seed"
+docker-compose exec -T seed-pg13 psql --user=seed -c 'GRANT ALL PRIVILEGES ON DATABASE "seed" TO "seed";'
+
+# Alter the user "seed" to have CREATEDB, CREATEROLE, and SUPERUSER privileges
+docker-compose exec -T seed-pg13 psql --user=seed -c 'ALTER USER "seed" CREATEDB CREATEROLE SUPERUSER;'
+
+# Create the PostGIS extension if it does not exist
+docker-compose exec -T seed-pg13 psql --user=seed -d seed -c 'CREATE EXTENSION IF NOT EXISTS postgis;'
+
+# Create the TimescaleDB extension if it does not exist
+docker-compose exec -T seed-pg13 psql --user=seed -d seed -c "CREATE EXTENSION IF NOT EXISTS timescaledb WITH VERSION '2.3.0';"
```

Comment:
This won't do anything without dropping the extension first:
`NOTICE:  extension "timescaledb" already exists, skipping`

Reply:
Good catch.